back/lib/tasks/single_use/20240910_migrate_initiatives_to_proposals.rake

```diff
@@ -0,0 +1,811 @@
+  previous_status_id = nil
+  initiative.initiative_status_changes.order(:created_at).each do |change|
+    next_status_id = rake_20240910_to_idea_status(change.initiative_status)&.id
+    activity = Activity.new(
+      item: proposal,
+      action: 'changed_status',
+      user: change.user,
+      acted_at: change.created_at,
+      payload: { change: [previous_status_id, next_status_id] }
+    )
+    previous_status_id = next_status_id
+    if !activity.save
+      reporter.add_error(
+        activity.errors.details,
+        context: { tenant: AppConfiguration.instance.host, proposal: proposal.slug, status_change: change.id }
+      )
+    end
+  end
+end
+
+def rake_20240910_migrate_images_files(proposal, initiative, reporter)
+  initiative.initiative_images.order(:ordering).each do |old_image|
+    new_image = IdeaImage.new(
+      idea: proposal,
+      remote_image_url: old_image.image_url
+    )
+    if !new_image.save
+      reporter.add_error(
+        new_image.errors.details,
+        context: { tenant: AppConfiguration.instance.host, proposal: proposal.slug, image: old_image.id }
+      )
+    end
+  end
+  initiative.initiative_files.order(:ordering).each do |old_file|
+    new_file = IdeaFile.new(
+      idea: proposal,
+      name: old_file.name,
+      remote_file_url: old_file.file_url
+    )
+    if !new_file.save
+      reporter.add_error(
+        new_file.errors.details,
+        context: { tenant: AppConfiguration.instance.host, proposal: proposal.slug, file: old_file.id }
+      )
+    end
+  end
+  initiative.text_images.each do |text_image|
+    text_image.imageable = proposal
+    if !text_image.save
+      reporter.add_error(
+        text_image.errors.details,
+        context: { tenant: AppConfiguration.instance.host, proposal: proposal.slug, text_image: text_image.id }
+      )
+    end
+  end
+end
+
+def rake_20240910_migrate_topics(proposal, initiative, reporter)
+  proposal.topics = initiative.topics
+  if !proposal.save
+    reporter.add_error(
+      proposal.errors.details,
+      context: { tenant: AppConfiguration.instance.host, proposal: proposal.slug }
+    )
+  end
+end
+
+def rake_20240910_migrate_reactions(proposal, initiative, reporter)
+  initiative.likes.each do |like|
+    reaction = Reaction.new(
+      mode: 'up',
+      user_id: like.user_id,
+      reactable: proposal,
+      created_at: like.created_at
+    )
+    if !reaction.save
+      reporter.add_error(
+        reaction.errors.details,
+        context: { tenant: AppConfiguration.instance.host, proposal: proposal.slug, reaction: like.id }
+      )
+    end
+  end
+end
+
+def rake_20240910_migrate_spam_reports(proposal, initiative, reporter)
+  initiative.spam_reports.each do |old_report|
+    new_report = SpamReport.new(
+      spam_reportable: proposal,
+      user_id: old_report.user_id,
+      reason_code: old_report.reason_code,
+      other_reason: old_report.other_reason,
+      created_at: old_report.created_at
```

Comment:
reported_at? Also updated_at?

Reply:
I seem to have overlooked `reported_at`. Will add both.